src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+        * `MeshVolumeRegion`
+        * `MeshSurfaceRegion`
+        * `PolygonalFootprintRegion`
+        """
+        # Check if region is fixed, and if not returns a default implementation
+        if needsSampling(self):
+            return super().intersects(other)
+
+        if isinstance(other, MeshVolumeRegion):
+            # PASS 1
+            # Check if bounding boxes intersect. If not, volumes cannot intersect.
+            # For bounding boxes to intersect there must be overlap of the bounds
+            # in all 3 dimensions.
+            range_overlaps = [(self.mesh.bounds[0,dim] <= other.mesh.bounds[1,dim]) and \
+                              (other.mesh.bounds[0,dim] <= self.mesh.bounds[1,dim]) \
+                              for dim in range(3)]
+            bb_overlap = all(range_overlaps)
+
+            if not bb_overlap:
+                return False
+
+            # PASS 2
+            # Compute inradius and circumradius for a candidate point in each region,
+            # and compute the inradius and circumradius of each point. If the candidate 
+            # points are closer than the sum of the inradius values, they must intersect.
+            # If the candidate points are farther apart than the sum of the circumradius
+            # values, they can't intersect.
+
+            # Get a candidate point from each mesh. If the center of the object is in the mesh use that.
+            # Otherwise try to sample a point as a candidate, skipping this pass if the sample fails.
+            if self.containsPoint(Vector(*self.mesh.bounding_box.center_mass)):
+                s_candidate_point = Vector(*self.mesh.bounding_box.center_mass)
+            elif len(samples:=trimesh.sample.volume_mesh(self.mesh, self.num_samples)) > 0:
+                s_candidate_point = Vector(*samples[0])
+            else:
+                s_candidate_point = None
+
+            if other.containsPoint(Vector(*other.mesh.bounding_box.center_mass)):
+                o_candidate_point = Vector(*other.mesh.bounding_box.center_mass)
+            elif len(samples:=trimesh.sample.volume_mesh(other.mesh, other.num_samples)) > 0:
+                o_candidate_point = Vector(*samples[0])
+            else:
+                o_candidate_point = None
+
+            if s_candidate_point is not None and o_candidate_point is not None:
+                # Compute the inradius of each object from its candidate point.
+                s_inradius = abs(trimesh.proximity.ProximityQuery(self.mesh).signed_distance([s_candidate_point])[0])
+                o_inradius = abs(trimesh.proximity.ProximityQuery(other.mesh).signed_distance([o_candidate_point])[0])
+
+                # Compute the circumradius of each object from its candidate point.
+                s_circumradius = numpy.max(numpy.linalg.norm(self.mesh.vertices - s_candidate_point, axis=1))
+                o_circumradius = numpy.max(numpy.linalg.norm(other.mesh.vertices - o_candidate_point, axis=1))
```

Comment:
It would be good to make the representative point and inradius/circumradius w.r.t. that point cached properties so that they don't need to be repeatedly calculated for fixed regions.

Reply:
Good point, added to the list.